### PR TITLE
Provide partial test suite for header validation

### PR DESCRIPTION
**Pull request by abailly**
This PR builds upon @AndrewWestberg's work on Praos headers validation, adding more tests in the form of JSON-formatted pairs of a context and a header, which may or may not be valid. A single test function loads the test vectors and iterates through the various headers, calling validation code and verifying the result. These tests increase code coverage from ~75% to ~95%.

In the course of writing these tests, I found a few minor issues I fixed in the validation code:

* Adding a special case for when the header's producer has 100% of the stake (which lead me to report [this issue](https://github.com/txpipe/pallas/issues/529)), which obviously is not a real case but is nevertheless accounted for in Haskell code and useful
* Using pool id instead of raw bytes as key to operational certificate counters' the consensus code maintains 
* Expose active slot coefficient from ledger instead of derived `c` value (ran the benchmarks locally and did not notice any significant impact), because of first point above
* Switch position for tests on operational counters as otherwise it could lead to runtime `panic` from underflow
* A couple more expressive error messages

There's more work that needs to be done to cover more of the validation process, which I plan to add in another PR:

* Vary stake distributions
* Cover VRF proof related checks

### CODE REVIEW

**Comment by abailly** [comment]
Consensus code PR: https://github.com/IntersectMBO/ouroboros-consensus/pull/1285

**Comment by AndrewWestberg** [comment]
Rebase on top of `main` since I merged my previous PR. It should be a minimal ff rebase for you.

File: ouroboros-praos/Cargo.toml [hunk 1]
```diff
@@ -6,14 +6,23 @@ edition = "2021"
...
 mockall = "0.13"
-pallas-traverse = { git = "https://github.com/txpipe/pallas", rev = "252714d58a93d58bf2cd7ada1156e3ace10a8298" }
-tracing-subscriber = "0.3"
\ No newline at end of file
+pallas-traverse = { git = "https://github.com/txpipe/pallas", rev = "7f988a16d412d4891408f99785372a4ef617984d" }
+tracing-subscriber = "0.3"
+criterion = "0.5.1"
+serde = { version = "1.0", features = ["derive"] }
+serde_json = "1.0.128"
+base64 = "0.22.1"
```
Comment (AndrewWestberg) [inline ouroboros-praos/Cargo.toml]: test vectors are not using base64 encoding any long. We can remove this dependency.

**Review by AndrewWestberg — APPROVED** [review]
Should be good to merge after rebasing. This is a good base to add additional tests for opcert mutations, etc on top of.

**Review by AndrewWestberg — CHANGES_REQUESTED** [review]
Ensure we are using conventional commit messages.

https://www.conventionalcommits.org/en/v1.0.0/#summary

**Comment by abailly** [comment]
@AndrewWestberg Can we merge this? It's been a while...

**Review by AndrewWestberg — APPROVED** [review]

**Pull request merged**